src/components/organisms/bucket-add-field/BucketAddField.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { FC, memo, useState } from "react";
+import { createShema } from "./BucketAddFieldSchema";
+import Button from "components/atoms/button/Button";
+import useInputRepresenter from "custom-hooks/useInputRepresenter";
+import Tab from "components/atoms/tab/Tab";
+import FlexElement from "components/atoms/flex-element/FlexElement";
+import FluidContainer from "components/atoms/fluid-container/FluidContainer";
+import Icon from "components/atoms/icon/Icon";
+import styles from "./BucketAddField.module.scss";
+import { TypeInputType } from "../../../custom-hooks/useInputRepresenter";
+import Modal from "components/atoms/modal/Modal";
+import { TypeModal } from "utils/interface";
+
+type TypeBucketAddField = {
+  name: string;
+  type: TypeInputType;
+  isOpen?: boolean;
+  modalProps?: TypeModal;
+};
+
+const BucketAddField: FC<TypeBucketAddField> = ({ name, type, isOpen, modalProps }) => {
+  const isInnerFieldsType = type === "object" || type === "array";
+  const initialTab = isInnerFieldsType ? 0 : 1;
+  const [activeTab, setActiveTab] = useState(initialTab);
+  const [configurationValue, setConfigurationValue] = useState({
+    definePattern: false,
+    primaryField: false,
+    translatable: false,
+    readonly: false,
+    uniqueValues: false,
+    requiredField: false,
+    selectionOptions: false,
+    index: false,
+  });
+  const configurationMapping = {
+    string: createShema.configuration,
+    number: createShema.configuration,
+    date: createShema.configurationType1,
+    color: createShema.configurationType1,
+    multiselect: createShema.configurationType1,
+    object: createShema.configurationType2,
+    storage: createShema.configurationType2,
+    richtext: createShema.configurationType2,
+    textarea: createShema.configurationTextarea,
+    boolean: createShema.configurationBoolean,
+    location: createShema.configurationLocation,
+    array: createShema.configurationArray,
+  };
+
+  const configurationSchema = configurationMapping[type] || {};
+  const schema = createShema[type] || {};
+  const schemaWithDynamicTitle = {
+    ...schema,
+    title: {
+      ...schema.title,
+      title: name || "",
+    },
+  };
+  const inputRepresenter = useInputRepresenter({
+    properties: schemaWithDynamicTitle,
+    value: {
+      title: name || "",
+      description: "",
+      ...Object.fromEntries(Object.keys(schema).map((key) => [key, ""])),
+    },
+    onChange: () => {},
+  });
+
+  const configuration = useInputRepresenter({
+    properties: configurationSchema,
+    value: configurationValue,
+    onChange: setConfigurationValue,
+  });
+
```

Comment:
It would be cleaner to define separate const for items.

```suggestion
const tabItems = [
  ...(isInnerFieldsType
    ? [
        {
          prefix: {
            children: "Inner Fields",
            onClick: () => setActiveTab(0),
          },
        },
      ]
    : []),
  {
    root: {
      children: "Configuration",
      onClick: () => setActiveTab(1),
    },
  },
  {
    suffix: {
      children: "Properties",
      onClick: () => setActiveTab(2),
    },
  },
];
```

Reply:
Done